src/platform/nrfconnect/OTAImageProcessorImpl.cpp

```diff
@@ -76,23 +66,59 @@ CHIP_ERROR OTAImageProcessorImpl::Apply()
 #endif
 }
 
+CHIP_ERROR OTAImageProcessorImpl::Abort()
+{
+    return System::MapErrorZephyr(dfu_target_reset());
+}
+
+CHIP_ERROR OTAImageProcessorImpl::Apply()
+{
+    int err = dfu_target_done(true);
+    if (err == 0)
+    {
+        err = dfu_target_schedule_update(-1);
+    }
+    return System::MapErrorZephyr(err);
+}
+
 CHIP_ERROR OTAImageProcessorImpl::ProcessBlock(ByteSpan & block)
 {
     VerifyOrReturnError(mDownloader != nullptr, CHIP_ERROR_INCORRECT_STATE);
 
     CHIP_ERROR error = ProcessHeader(block);
-
     if (error == CHIP_NO_ERROR)
     {
-        // DFU target library buffers data internally, so do not clone the block data.
-        error = System::MapErrorZephyr(dfu_target_write(block.data(), block.size()));
+        mParams.downloadedBytes += block.size();
+        if (mParams.downloadedBytes >= mContentHeader.mFiles[mCurrentImage].mFileSize)
```

Comment:
I think you may want to accumulate the file sizes and calculate _file end offsets_ for each file. Otherwise, this condition will be needlessly triggered just after downloading the first block of the network image, right?